document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/SmallFullAccountsDataHandler.java

```diff
@@ -0,0 +1,158 @@
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.model.accounts.CompanyAccounts;
+import uk.gov.companieshouse.document.generator.accounts.AccountType;
+import uk.gov.companieshouse.document.generator.accounts.LinkType;
+import uk.gov.companieshouse.document.generator.accounts.data.transaction.Transaction;
+import uk.gov.companieshouse.document.generator.accounts.exception.HandlerException;
+import uk.gov.companieshouse.document.generator.accounts.exception.ServiceException;
+import uk.gov.companieshouse.document.generator.accounts.mapping.smallfull.model.ixbrl.SmallFullAccountIxbrl;
+import uk.gov.companieshouse.document.generator.accounts.service.AccountsService;
+import uk.gov.companieshouse.document.generator.accounts.service.CompanyService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.document.generator.accounts.AccountsDocumentInfoServiceImpl.MODULE_NAME_SPACE;
+
+@Component
+public class SmallFullAccountsDataHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
+
+    @Autowired
+    private AccountsService accountsService;
+
+    @Autowired
+    private CompanyService companyService;
+
+    private static final DateFormat RESPONSE_DISPLAY_DATE_FORMAT = new SimpleDateFormat("dd MMMMM yyyy");
+
+    private static final DateFormat ISO_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+
+    private static final String RESOURCE = "resource";
+
+    private static final String ACCOUNT_TYPE = "accountType";
+
+    /**
+     * Get a SmallFull Accounts resource from the given resource link
+     *
+     * @param transaction the transaction data
+     * @param resourceLink the resource link of the smallFull accounts
+     * @param requestId the id of the request
+     * @return a populated {@link DocumentInfoResponse} object
+     * @throws HandlerException throws a custom handler exception
+     */
+    public DocumentInfoResponse getSmallFullAccountsData(Transaction transaction, String resourceLink, String requestId)
+            throws HandlerException {
+
+        CompanyAccounts accounts = getCompanyAccounts(resourceLink, requestId);
+
+        AccountType accountType = getCompanyAccountType(accounts);
+
+        String smallFullAccountLink = getCompanyAccountLink(accounts, accountType);
+
+        try {
+            SmallFullAccountIxbrl smallFullAccountIxbrl = accountsService.getSmallFullAccounts(smallFullAccountLink, resourceLink, transaction);
+            return createResponse(accountType, smallFullAccountIxbrl);
+        } catch (ServiceException | IOException e) {
```

Comment:
Could we wrap the `IOException` as a `ServiceException` in the accounts service? The handler shouldn't have any knowledge that the service has performed input / output operations

Reply:
the IOException is thrown when return mapper.writeValueAsString(accountData); as part of the method createDocumentInfoResponseData() within the smallFullDataHandler

Reply:
Sorry, I mean can we catch the IOException and wrap it in a ServiceException and throw that from the service? The handler shouldn't care what type of exception occurred, only that an exception did occur in the service.